lib/Dialect/RNS/IR/RNSOps.td

```diff
@@ -3,11 +3,30 @@
 
 include "lib/Dialect/RNS/IR/RNSDialect.td"
 include "lib/Dialect/RNS/IR/RNSTypes.td"
+include "mlir/IR/BuiltinAttributes.td"
 include "mlir/IR/OpBase.td"
+include "mlir/Interfaces/InferTypeOpInterface.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
 
 class RNS_Op<string mnemonic, list<Trait> traits = []> :
         Op<RNS_Dialect, mnemonic, traits> {
   let cppNamespace = "::mlir::heir::rns";
 }
 
+def RNS_ExtractSliceOp : RNS_Op<"extract_slice", [Pure, ElementwiseMappable, DeclareOpInterfaceMethods<InferTypeOpInterface, ["inferReturnTypes"]>]> {
```

Comment:
We probably want a basic verifier for this (start is in range, start+size in range, size > 0, etc). Maybe just add a TODO for now, though?

Reply:
Done